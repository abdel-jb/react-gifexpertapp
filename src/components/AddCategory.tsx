import React from 'react';

type TAddCategory = {
    setCategories: React.Dispatch<React.SetStateAction<string[]>> | any;
}

export const AddCategory = ({ setCategories }: TAddCategory): JSX.Element => {

    const [inputValue, setInputValue] = React.useState('');

    const handleInputChanhe = (e: React.FormEvent<HTMLInputElement>): void => {
        const { value } = e.target as HTMLInputElement;
        setInputValue(value);
    }

    const handleSubmit = (e: React.FormEvent): void => {
        e.preventDefault();
        if (inputValue.trim().length > 2) {
            setCategories(( category: string[] ) => (!category.find(e => e === inputValue) ? [inputValue, ...category] : category));
            setInputValue('');
        }
    };

    return (
        <form action="" onSubmit={ handleSubmit }>
            <input 
                type="text"
                value={ inputValue }
                onChange={ handleInputChanhe }
            />
        </form>
    )
}
