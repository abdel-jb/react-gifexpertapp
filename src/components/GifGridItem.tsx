import React from 'react';

import { TGifs } from '../utils/types';


export const GifGridItem = ( { title, url }: TGifs ): JSX.Element => {
    return (
        <li className="card animate__animated animate__fadeIn">
            <img src={ url } alt={ title }/>
            <p>{ title }</p>
        </li>
    )
}
