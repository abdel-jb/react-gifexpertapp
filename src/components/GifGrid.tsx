import React from 'react';

import { GifGridItem } from './GifGridItem';
import { TGifs } from '../utils/types';
import { useFetchGifs } from '../hooks/useFetchGifs';


type TGifGrid = {
    category: string
}

export const GifGrid = ({ category }: TGifGrid): JSX.Element => {

    const { data: images, loading } = useFetchGifs(category);

    return (
        <>
            <h3>{ category }</h3>

            { loading && <p>Loading...</p> }

            <ul className="card-grid">
                { images.map(( img: TGifs ) => 
                    <GifGridItem
                        key={ img.id }
                        { ...img }
                    />
                )}
            </ul>
        </>
    )
}
