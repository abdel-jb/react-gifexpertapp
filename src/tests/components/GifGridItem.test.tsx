import React from 'react';
import { shallow } from 'enzyme';

import { GifGridItem } from '../../components/GifGridItem';

describe('Tests in <GifGridItem />', () => {

    const props = {
        title: 'opm GIF',
        url: 'https://media0.giphy.com/media/oxbNORcXx76F2/giphy.gif?cid=41ddacc3jrujn7t2slz1g1f4ul0osaaytjq921so484i51cy&rid=giphy.gif&ct=g'
    }
    const wrapper = shallow(<GifGridItem {...props}/>);

    test('should show the component correctly', () => {

        expect( wrapper ).toMatchSnapshot();
    });

    test('should have a paragraph with a title', () => {
        
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( props.title );
    });

    test('should have a image equal to the url and alt of props', () => {
        
        const img = wrapper.find('img');
        
        expect( img.prop('src') ).toBe( props.url );
        expect( img.prop('alt') ).toBe( props.title );

    });

    test('should have animate__fadeIn', () => {

        const li = wrapper.find('li.animate__fadeIn');

        expect( li.hasClass('animate__fadeIn') ).toBe( true );
 
    });
    
    
    
});
