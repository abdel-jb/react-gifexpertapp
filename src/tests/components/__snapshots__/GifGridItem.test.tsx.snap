// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <GifGridItem /> should show the component correctly 1`] = `
<li
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="opm GIF"
    src="https://media0.giphy.com/media/oxbNORcXx76F2/giphy.gif?cid=41ddacc3jrujn7t2slz1g1f4ul0osaaytjq921so484i51cy&rid=giphy.gif&ct=g"
  />
  <p>
    opm GIF
  </p>
</li>
`;
